src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -454,6 +467,53 @@ public bool TryCreateType(ITypeSymbol symbol, Func<System.Type, bool> validator,
             return true;
         }
 
+        private bool TryCreateTypeForIArrayTypeSymbol(ITypeSymbol symbol, [NotNullWhen(true)] out CSharpType? type)
+        {
+            type = null;
+            if (symbol is not IArrayTypeSymbol arrayTypeSymbol)
+            {
+                return false;
+            }
+
+            // For IArrayTypeSymbol, we can only handle it when the element type is a framework type.
+            var arrayType = TryGetFrameworkType(arrayTypeSymbol);
+            if (arrayType is not null)
```

Comment:
I believe we also should have the validator here as well

Reply:
yeah, sorry forgot it again...